gtsfm/loader/astronet_loader.py

```diff
@@ -187,8 +190,18 @@ def get_image_full_res(self, index: int) -> Image:
         if index < 0 or index >= len(self):
             raise IndexError(f"Image index {index} is invalid")
 
+        # Read in image.
         img = io_utils.load_image(self._image_paths[index])
-        return img
+
+        # Generate mask to separate background deep space from foreground target body
+        # based on image intensity values.
+        gray_image = image_utils.rgb_to_gray_cv(img)
+        _, binary_image = cv.threshold(gray_image.value_array, 5, 255, cv.THRESH_BINARY)
```

Comment:
let's make this a standalone function. Perhaps Image.py would be a good place for it. But here is ok too, if you think it's too domain specific to be useful elsewhere.

```python
def get_nonzero_intensity_mask(img: Image) -> np.ndarray:
      """Generate mask of where image intensity values are non-zero.
      
      Args:
            img: ...
      """
       gray_image = image_utils.rgb_to_gray_cv(img)
       _, binary_image = cv.threshold(gray_image.value_array, 5, 255, cv.THRESH_BINARY)
       mask = cv.erode(binary_image, np.ones((15, 15), np.uint8))
       return mask
```

Reply:
keyword args on the 15x15 kernel and for the params 5 and 255 would also help make it even more readable

Reply:
Yeah I'm not sure sure what other instances it could be used for; think it just makes sense to keep it here. 

Yes I should make those keyword args.

Reply:
Sure, we can keep it here. Let's make it a separate function though, to improve readability.

Reply:
moved to separate function and added keyword args